src/main/java/tkachuk/paint/PaintFrame.java

```diff
@@ -114,34 +141,40 @@ public void mouseClicked(MouseEvent event)
             @Override
             public void mousePressed(MouseEvent event)
             {
-                Graphics g = canvas.getImage().getGraphics();
-                g.setColor(Color.BLACK);
-                pencilTool.pressed(g, event.getX(), event.getY());
-                canvas.repaint();
-
-//                if (linePressed)
-//                {
-//                    start = event.getPoint();
-//                    canvas.setStartPoint(start);
-//                } else if (pencilPressed)
-//                {
-//                    canvas.setOldxy(event.getX(), event.getY());
-//                }
+
+                if (linePressed)
+                {
+                    Graphics g = canvas.getImage().getGraphics();
+                    g.setColor(canvas.getDrawColor());
+                    lineTool.pressed(g, event.getX(), event.getY());
+                    canvas.repaint();
+                } else if (pencilPressed)
+                {
+                    Graphics g = canvas.getImage().getGraphics();
+                    g.setColor(canvas.getDrawColor());
+                    pencilTool.pressed(g, event.getX(), event.getY());
+                    canvas.repaint();
+                } else if (eraserPressed)
+                {
+                    Graphics g = canvas.getImage().getGraphics();
+                    g.setColor(Color.WHITE);
+                    eraserTool.pressed(g, event.getX(), event.getY());
+                    canvas.repaint();
+                }
```

Comment:
Can you think of a better way of doing this so you don't have all this code repeating?